### DISCUSSION TITLE
How to make description column the 2nd column ?

### DISCUSSION THREAD

**Discussion by towry** [mrjones2014/legendary.nvim#290] [Q&A] [answered]
![截屏2022-12-27 17 47 59](https://user-images.githubusercontent.com/8279858/209647918-ad9f0964-de9c-4cd1-95ab-3c0fe93a1801.png)

While I typing a keyword, I will check the column to see if the keyword is sufficient enough. It is strange/uncomfortable that I have to move my eye from the left(where the keyword at) to the right(description column), see the above picture.

If the description is at 2nd column, I just need to move eye vertically, usually just the first line below the keyword, it is more convenient, what do you think?

**Comment by mrjones2014** [comment, marked as answer]
You can override the columns via `config.default_item_formatter` which takes an item and returns a list of strings, one string per column. Here's the default formatter code:

```lua
---Default format
---@param item LegendaryItem
---@return string[]
function M.default_format(item)
  if Toolbox.is_keymap(item) then
    return {
      table.concat(item:modes(), ', '),
      item.keys,
      item.description,
    }
  elseif Toolbox.is_command(item) then
    return {
      '',
      item.cmd,
      item.description,
    }
  elseif Toolbox.is_autocmd(item) then
    return {
      table.concat(item.events, ', '),
      table.concat(vim.tbl_get(item, 'opts', 'pattern') or { '*' }, ', '),
      item.description,
    }
  elseif Toolbox.is_function(item) then
    return {
      '',
      '<function>',
      item.description,
    }
  elseif Toolbox.is_itemgroup(item) then
    return {
      item.icon or '',
      item.name,
      item.description or 'Expand to select an item...',
    }
  else
    -- unreachable
    return {
      vim.inspect(item),
      '',
      '',
    }
  end
end
```

So you could write a function that's basically the same but swaps the 2nd and 3rd columns, then assign it like:


```lua
require('legendary').setup({ default_item_formatter = my_format_fn })
```

This is described briefly in the README.md as well under [Configuration](https://github.com/mrjones2014/legendary.nvim#configuration).

**Reply by mrjones2014** [reply]
Actually, as a little trick to make your formatter simpler, you could make your formatter function this:

```lua
local function my_formatter(item)
  local default_columns = require('legendary.ui.format').default_format(item)
  local swap = default_columns[2]
  default_columns[2] = default_columns[3]
  default_columns[3] = swap
end
```
